docs/use-cases/cd-autotests-run-with-env-variables.md

```diff
@@ -0,0 +1,304 @@
+- The ability to manage environment variables by overriding, adding, or removing their values and keys to configure test parameters.
+- A streamlined setup process that allows teams to focus on development without platform constraints.
+
+## Preconditions
+
+- KubeRocketCI instance with GitHub and Tekton is [configured](../operator-guide/prerequisites.md).
+- Developer has access to the KubeRocketCI instances using the Single-Sign-On approach.
+- Developer has `Write` permissions for GitHub repository to merge the code.
+
+## Scenario
+
+To implement custom logic in deployment flow quality gate within the KubeRocketCI platform, follow these steps:
+
+- **Fork the Custom Tekton Chart**: Fork, clone and deploy the custom Tekton chart that implements the required Tekton resources with the environment variables usage.
+- **Add the Application**: Add a simple application from the platform marketplace and build the application to prepare it for deployment.
+- **Add the Autotests**: Set up the autotests with the clone strategy to include them in the Deployment Flow.
+- **Create Deployment Flow, Environment, and Set Variables**: Create a Deployment Flow, Set up an environment for the deployment, add Autotests as a quality gate to the flow, and define the variable.
+- **Validate**: Review the pipeline logs to ensure autotests execute correctly and produce the expected output.
+
+### Fork the Custom Tekton Chart
+
+The first step is to clone and install the custom Tekton chart:
+
+1. Create a private repository and copy the [chart repository](https://github.com/alezander86/custom-tekton-chart) content. Below is the repository structure:
+
+    <details>
+    <summary>Repository structure</summary>
+    ```bash
+      custom-tekton-chart
+        .
+        ├── Chart.lock
+        ├── Chart.yaml
+        ├── README.md
+        ├── chart_schema.yaml
+        ├── ct.yaml
+        ├── lintconf.yaml
+        ├── templates
+        │   ├── pipelines
+        │   │   └── cd
+        │   │       ├── autotest-variables.yaml
+        │   │       └── deploy-with-autotests-variables.yaml
+        │   ├── tasks
+        │   │   └── cd
+        │   │       ├── init-autotests-variables.yaml
+        │   │       └── run-autotests-variables-java.yaml
+        │   └── triggers
+        │       └── cd
+        │           └── deploy-with-autotests-variables.yaml
+        └── values.yaml
+    ```
+    </details>
+
+2. Clone your private repository to your local machine:
+
+  ![Clone repo](../assets/use-cases/cd-autotests-run-with-env-variables/clone-repo.png "Clone repo")
+
+  ```bash
+  git clone git@github.com:<github_account_name>/custom-tekton-chart.git
+  ```
+
+3. Update the component configuration in the `values.yaml` file:
+
+    ```yaml
+    global:
+      # Replace `example.domain.com` with the appropriate wildcard domain for your setup.
+      dnsWildCard: "example.domain.com"
+    ```
+
+4. Update Helm dependencies:
```

Comment:
do we need this dependency? in upstream we do not use this approach 
https://github.com/epmd-edp/helm-helm-pipeline/blob/master/Chart.yaml